rllib/policy/view_requirement.py

```diff
@@ -0,0 +1,55 @@
+from gym.spaces import Box, Space
+from typing import Optional
+
+from ray.rllib.utils.framework import try_import_torch
+
+torch, _ = try_import_torch()
+
+
+class ViewRequirement:
+    """Single view requirement (for one column in a ModelV2 input_dict).
+
+    Note: This is an experimental class.
+
+    ModelV2 returns a Dict[str, ViewRequirement] upon calling
+    `ModelV2.get_view_requirements()`, where the str key represents the column
+    name (C) under which the view is available in the `input_dict` and
+    ViewRequirement specifies the actual underlying column names (in the
+    original data buffer), timesteps, and other options to build the view
+    for N.
+
+    Examples:
+        >>> # The default ViewRequirement for a Model is:
+        >>> req = [ModelV2].get_view_requirements(is_training=False)
+        >>> print(req)
+        {"obs": ViewRequirement(timesteps=0)}
+    """
+
+    def __init__(self,
+                 data_col: Optional[str] = None,
+                 space: Space = None,
+                 shift: int = 0,
+                 sampling: bool = True,
```

Comment:
Let's try to remove these "when is the view requirement used" flags from the view req. Instead, we should keep a data structure that has separate lists of which view requirements are used in which scenarios.

Reply:
ok, see my answer above, too on the "when is the view requirement used" flags.